ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/ParamModel.java

```diff
@@ -0,0 +1,310 @@
+        }
+
+        /**
+         * Bean parameter identifier.
+         *
+         * @param beanParam {@link BeanParam} annotation
+         * @return updated Builder instance
+         */
+        Builder beanParam(BeanParam beanParam) {
+            this.beanParam = beanParam != null;
+            return this;
+        }
+
+        /**
+         * Cookie parameter.
+         *
+         * @param cookieParam {@link CookieParam} annotation
+         * @return updated Builder instance
+         */
+        Builder cookieParam(CookieParam cookieParam) {
+            this.cookieParamName = cookieParam == null ? null : cookieParam.value();
+            return this;
+        }
+
+        /**
+         * Query parameter.
+         *
+         * @param queryParam {@link QueryParam} annotation
+         * @return updated Builder instance
+         */
+        Builder queryParam(QueryParam queryParam) {
+            this.queryParamName = queryParam == null ? null : queryParam.value();
+            return this;
+        }
+
+        /**
+         * Matrix parameter.
+         *
+         * @param matrixParam {@link MatrixParam} annotation
+         * @return updated Builder instance
+         */
+        Builder matrixParam(MatrixParam matrixParam) {
+            this.matrixParamName = matrixParam == null ? null : matrixParam.value();
+            return this;
+        }
+
+        /**
+         * Form parameter.
+         *
+         * @param formParam {@link FormParam} annotation
+         * @return updated Builder instance
+         */
+        Builder formParam(FormParam formParam) {
+            this.formParamName = formParam == null ? null : formParam.value();
+            return this;
+        }
+
+        /**
+         * Position of parameter in method parameters
+         *
+         * @param paramPosition Parameter position
+         * @return updated Builder instance
+         */
+        Builder paramPosition(int paramPosition) {
+            this.paramPosition = paramPosition;
+            return this;
+        }
+
+        /**
+         * Returns path param name;
+         *
+         * @return path param name
+         */
+        String pathParamName() {
+            return pathParamName;
+        }
+
+        /**
+         * Returns header param name;
+         *
+         * @return header param name
+         */
+        String headerParamName() {
+            return headerParamName;
+        }
+
+        String cookieParamName() {
+            return cookieParamName;
+        }
+
+        String queryParamName() {
+            return queryParamName;
+        }
+
+        String matrixParamName() {
+            return matrixParamName;
+        }
+
+        String formParamName() {
+            return matrixParamName;
+        }
+
+        /**
+         * Creates new ParamModel instance.
+         *
+         * @return new instance
+         */
+        ParamModel build() {
+            if (pathParamName != null) {
```

Comment:
Could you create the param model when the parameter is set?
e.g.
```
Builder formParam(FormParam formParam) {
   if (null != formParam) {
    this.paramModel = new FormParamModel(formParam);
  } 
return this
 }
```
And then build method would just return the this.paramModel, and if null do the entity trick?